### DISCUSSION TITLE
Having multiple edges pointing to one single handle, makes it hard for the user to update an edge from a handle to another

### DISCUSSION THREAD

**Discussion by zakaria-imzilen** [xyflow/xyflow#4183] [General]
### Describe the Bug

Considering having 3 nodes, 2 of them have 2 edges pointing to the 3rd node on one single handle, 
Changing the 2 edge handles is impossible by selecting and dragging it to another handle.

In this example below (image), if I'd like to change the top edge from pointing out to the "Delivery and Customer Support" node to another one, I'd like to select the edge between them and then try to drag its end to another node. This approach doesn't always work.

There's some other logic behind that I don't understand.

<img width="871" alt="Screenshot 2024-04-18 at 12 33 56" src="https://github.com/xyflow/xyflow/assets/66393914/7f7021c9-bbad-4c91-bcb3-954fe338a0d6">

### Your Example Website or App

https://flowsiti.ai

### Steps to Reproduce the Bug or Issue

1- Click on Start Here
2- Click on the "First Process" button
3- Drag and drop a handle to create an adjacent node (Repeat 2 times) to end up with the same image I attached
3- Now try to change one of the edges target handle.

### Expected behavior

As a user, I expect when selecting an edge to be able to move it anywhere else.

### Screenshots or Videos

<img width="871" alt="Screenshot 2024-04-18 at 12 42 31" src="https://github.com/xyflow/xyflow/assets/66393914/23f3eb61-ee7b-4d44-aae3-109b9d6c80a3">

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 91.1]

**Comment by moklick** [comment]
This behaviour is not built-in. Where do you want to move the edge?

**Reply by zakaria-imzilen** [reply]
I just want to be able to select the wanted edge between those 2 AND move it somewhere else.

The issue is having the difficulty to select the desired edge.
